src/clients/repositories.rb

```diff
@@ -1683,6 +1683,9 @@ def SummaryDialog
               Ops.set(sourceState, "priority", new_priority)
               Ops.set(@sourceStatesOut, global_current, sourceState)
 
+              # update the priority for the already loaded packages
+              Pkg.SourceSetPriority(sourceState["SrcId"], new_priority)
```

Comment:
Updated, the same problem was actually also when enabling/disabling a repository.